### PR TITLE
Adds support for defining specific account name for AWS Service Catalog

### PR DESCRIPTION
**Pull request by fgoncalves-io**
The current approach to define the account name in the `avm` module is not flexible which blocks migrating existing organizations to this module.

This PR proposes the following changes:
- Adding a variable called `prefixed_name` that indicates whether the account name should be prefixed or not instead of always adding the prefix (default value is `true`). We already have use cases of accounts that were provisioned via Control Tower with a non-prefixed name. So this would allow us to migrate those accounts to the `avm` module without issues.
- Only appending the environment to the account name in case it's different than `prod`.
- Standardizing the account alias and the name passed to the terraform-aws-mcaf-account module
- Adds an extra output related to the KMS key created in the master account

### CODE REVIEW

**Comment by shoekstra** [comment]
This needs some conversation before we merge...

**Comment by fgoncalves-io** [comment]
@shoekstra, I've checked Service Catalog and we can update the provisioned product name there which should solve the issue the `prefixed_name` variable was trying to solve when importing existing accounts.

So I've removed the variable and the change now is only about the following points: 
- Only appending the environment to the account name in case it's different than prod.
- Standardizing the account alias and the name passed to the terraform-aws-mcaf-account module
- Adds an extra output related to the KMS key created in the master account

File: modules/avm/main.tf [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+  email               = var.email != null ? var.email : "${var.defaults.account_prefix}-aws-${local.name}@schubergphilis.com"
+  name                = var.environment == null || var.environment == "prod" ? var.name : "${var.name}-${var.environment}"
   organizational_unit = var.organizational_unit != null ? var.organizational_unit : var.environment == "prod" ? "Production" : "Non-Production"
+  prefixed_name       = "${var.defaults.account_prefix}-${local.name}"
 }
 
 module "account" {
   source                   = "github.com/schubergphilis/terraform-aws-mcaf-account?ref=v0.3.0"
-  account                  = local.name
+  account                  = local.prefixed_name
```
Comment (shoekstra) [inline modules/avm/main.tf:10]: As discussed, let's revert these changes and add an `account_name` variable that will be used here if it's set:

```suggestion
  account                  = coalesce(var.account_name, local.prefixed_name)
```

Rest of the proposed changes can be reverted.

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: modules/avm/main.tf (hunk 1 above)
Comment (fgoncalves-io) [reply modules/avm/main.tf]: I've used `coalesce(var.account_name, local.name)` instead of `coalesce(var.account_name, local.prefixed_name)` to keep it backwards compatible and to also follow our discussion regarding prefixes not been really necessary.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## Unreleased
+
+* Adds support for defining specific account name for AWS Service Catalog ([#13](https://github.com/schubergphilis/terraform-aws-mcaf-landing-zone/pull/13))
```
Comment (shoekstra) [inline CHANGELOG.md:9]:
```suggestion
* Add support for defining specific account name for AWS Service Catalog ([#13](https://github.com/schubergphilis/terraform-aws-mcaf-landing-zone/pull/13))
```

**Review by shoekstra — CHANGES_REQUESTED** [review]

**Review by shoekstra — APPROVED** [review]
